src/wgl_context.c

```diff
@@ -321,10 +321,15 @@ static void swapBuffersWGL(_GLFWwindow* window)
     {
         if (IsWindowsVistaOrGreater())
         {
-            BOOL enabled;
+            BOOL enabled = FALSE;
+          
+            // DWM Composition is always enabled on Win8+
+            if (IsWindows8OrGreater())
```

Comment:
Instead of doing this check, you can just initialize `enabled` with this call
```
BOOL enabled = IsWindows8OrGreater();
```

Reply:
An even better solution would be to change the initial condition
```
if (IsWindowsVistaOrGreater() && !IsWindows8OrGreater())
```

Reply:
Good call on the initialization, that's much better. I don't think I can do your second suggestion though because these blocks are intended to run when dwm composition is enabled, so they definitely should execute on Win8+.

Reply:
My bad, sorry. I got confused, thought we're skipping this part.